### ISSUE TITLE
scope annotation should be `sealedsecrets.bitnami.com/scope`

### ISSUE THREAD

**Issue by ghostsquad** [bitnami-labs/sealed-secrets#391]
It doesn't make sense to combine the following annotations:

```
sealedsecrets.bitnami.com/cluster-wide
sealedsecrets.bitnami.com/strict
sealedsecrets.bitnami.com/namespace-wide
```

so the annotation for this should really be:

```
sealedsecrets.bitnami.com/scope
```

with the possible values being `cluster-wide`, `strict`, `namespace-wide`

**Comment by mkmik** [comment]
Yes, there is a todo in the code too. The only reason it's not done yet is lack of time (it has to be done in a backward compatible way). Contributiona welcome!

**Comment by mkmik** [comment]
Readme says:

> NOTE: next release will consolidate this into a single sealedsecrets.bitnami.com/scope annotation.

:-) unfortunately it's been a few releases since that promise has been made :-)

On the meantime though we now support the --scope flag on kubeseal, which means you're less likely to be required to fiddle with those annotation s manually.

**Comment by ghostsquad** [comment]
ya, it was just something where if I was writing code and looking at annotations, it's something where I'd have to check several keys instead of checking one known key and collecting its value.

**Comment by ghostsquad** [comment]
Any change on this front? Simply supporting both annotations would make this backwards compatible. I'm happy to contribute. It sounds like this is a desired feature.

**Comment by mkmik** [comment]
I'd like to have this. Contributions are welcome!

Small PRs with tests are more likely to be merged quickly :-)

**Comment by mkmik** [comment]
Also, shouldn't we encode the scope in a real field of the SealedSecret resource instead of an annotation?

**Comment by ghostsquad** [comment]
that's a good call. Usually annotations are used when resources need to interact with other resources in a "generic" way, without requiring the need of a CRD. In this case the sealedSecret is already a CRD, and so really it should just be a field.

**Comment by mkmik** [comment]
We still need to have the annotation in the Secret resource though.

**Comment by ghostsquad** [comment]
@mkmik can you help me understand what the interaction is between sealed secret and vanilla secret that needs the annotation on tht vanilla?

**Comment by ghostsquad** [comment]
Thank you for the history! It clarifies a lot. This still seems doable, it depends on whether or not kubeseal reads the annotation on the sealed secret or just uses the CLI flag and/or vanilla secret annotation.

If the latter, changing to a real CRD field should be backwards compatible while still supporting roundtrip functionally.

**Comment by mkmik** [comment]
Kubeaeal reads the annotation and the flag. If both are set, the flag takes precedence
